### DISCUSSION TITLE
Can't access TrueNAS host from a docker container running in a jail

### DISCUSSION THREAD

**Discussion by pheitman** [Jip-Hop/jailmaker#211] [General]
I am trying to get homepage running on docker within my docker jail I have created on TrueNAS scale. I have lots of apps that work fine but this is the first time I want to access the TrueNAS URL from within a container. The TrueNAS IP address is unreachable from within the container. I have tried the default container network, host and bridge to no avail. I have also tried the host.docker.internal approach. Any ideas for what I need to do so that the TrueNAS port on the host is accessible?

**Comment by Jip-Hop** [comment]
We'd be able to help you better if you post more info such as your config. Also read this: https://github.com/Jip-Hop/jailmaker#filing-issues-and-community-support

**Reply by pheitman** [reply]
That's fair!

my jlmkr edit docker config follows. I think that the only change I've made is the --capabilities=all. I needed that to be able to run paperless-ngx which required CAP_IPC_LOCK. As I mentioned, I have tried several things to be able to access the host from within the container. network_mode=host, network_mode=bridge, the default container network (which was my default choice) and adding host.docker.internal as an extra_host. In all of these cases the host is unreachable. 

```
startup=1
gpu_passthrough_intel=0
gpu_passthrough_nvidia=0
# Turning off seccomp filtering improves performance at the expense of security
seccomp=1

# Use macvlan networking to provide an isolated network namespace,
# so docker can manage firewall rules
# Alternatively use --network-macvlan=eno1 instead of --network-bridge
# Ensure to change eno1/br1 to the interface name you want to use
# You may want to add additional options here, e.g. bind mounts
systemd_nspawn_user_args=--network-macvlan=eno1
        --resolv-conf=bind-host
        --system-call-filter='add_key keyctl bpf'
        --bind='/mnt:/mnt'
        --capability=all

# Script to run on the HOST before starting the jail
# Load kernel module and config kernel settings required for docker
pre_start_hook=#!/usr/bin/bash
        set -euo pipefail
        echo 'PRE_START_HOOK'
        echo 1 > /proc/sys/net/ipv4/ip_forward
        modprobe br_netfilter
        echo 1 > /proc/sys/net/bridge/bridge-nf-call-iptables
        echo 1 > /proc/sys/net/bridge/bridge-nf-call-ip6tables

# Only used while creating the jail
distro=debian
release=bookworm

# Install docker inside the jail:
# https://docs.docker.com/engine/install/debian/#install-using-the-repository
# Will also install the NVIDIA Container Toolkit if gpu_passthrough_nvidia=1 during initial setup
# https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html
initial_setup=#!/usr/bin/bash

```

**Reply by Jip-Hop** [reply]
That's a limitation of using macvlan networking. It's documented in the networking docs in the jailmaker repo. Please use bridge networking if you need to communicate with the TrueNAS host.

**Reply by Jip-Hop** [reply]
P.S. If you only need `CAP_IPC_LOCK` then there's no need to pass all capabilities.

**Reply by pheitman** [reply]
I tried just passing in CAP_IPC_LOCK and all of my containers stopped working. I must have done something wrong, but passing in all capabilities 'just worked' (with any downsides hidden from me).

I'll look in to using bridge networking ASAP. Thanks for the help!

**Reply by Jip-Hop** [reply]
Restricting capabilities to only the required ones helps with isolation and therefore increases security. That would be the benefit of not passing all capabilities.

**Reply by pheitman** [reply]
I was able to set up bridge networking (following the youtube video and the documentation on this github site). I was pleased that I was able to set it up with the static IP address that my DHCP server had assigned to the TrueNAS instance. That meant fewer headaches because some things (like my Windows hosts file) already had that address configured. After setting up the bridge I was able to configure the service widgets in Homepage that accessed the containers via ports on the docker jail. I will look in to only passing in the capabilities I need next. Thanks again for the help.
